python/ray/_private/job_manager/job_manager.py

```diff
@@ -207,64 +246,88 @@ def _get_current_node_resource_key(self) -> str:
+                   _start_signal_actor: Optional[SignalActor] = None) -> str:
         """
-        1) Create new detached actor with same runtime_env as job spec
-        2) Get task / actor level runtime_env as env var and pass into
+        Job execution happens asynchronously.
+
+        1) Generate a new unique id for this job submission, expected to be
+            executed in complete isolation.
+        2) Create new detached actor with same runtime_env as job spec
+        3) Get task / actor level runtime_env as env var and pass into
             subprocess
-        3) subprocess.run(entrypoint)
+        4) asyncio.create_subprocess_shell(entrypoint)
+
+        Args:
+
+        Returns:
 
-        Returns unique job_id.
         """
         job_id = str(uuid4())
-        supervisor = self._supervisor_actor_cls.options(
-            lifetime="detached",
-            name=self.JOB_ACTOR_NAME.format(job_id=job_id),
-            # Currently we assume JobManager is created by dashboard server
-            # running on headnode, same for job supervisor actors scheduled
-            resources={
-                self._get_current_node_resource_key(): 0.001,
-            },
-            # For now we assume supervisor actor and driver script have same
-            # runtime_env.
-            runtime_env=runtime_env,
-        ).remote(job_id, metadata or {})
+        self._status_client.put_status(job_id, JobStatus.PENDING)
+        supervisor = None
 
         try:
+            logger.debug(
+                f"Submitting job with generated internal job_id: {job_id}")
+            supervisor = self._supervisor_actor_cls.options(
+                lifetime="detached",
+                name=self.JOB_ACTOR_NAME.format(job_id=job_id),
+                # Currently we assume JobManager is created by dashboard server
+                # running on headnode, same for job supervisor actors scheduled
+                resources={
+                    self._get_current_node_resource_key(): 0.001,
+                },
+                # For now we assume supervisor actor and driver script have
+                # same runtime_env.
+                runtime_env=runtime_env).remote(job_id, metadata or {})
             ray.get(
                 supervisor.ready.remote(), timeout=self.START_ACTOR_TIMEOUT_S)
-        except GetTimeoutError:
-            ray.kill(supervisor, no_restart=True)
-            raise RuntimeError(f"Failed to start actor for job {job_id}.")
+        except Exception as e:
+            if supervisor:
+                ray.kill(supervisor, no_restart=True)
+            self._status_client.put_status(job_id, JobStatus.FAILED)
+            raise RuntimeError(
+                f"Failed to start actor for job {job_id}. This could be "
+                "runtime_env configuration failure, or timed out after "
+                f"{self.START_ACTOR_TIMEOUT_S} secs. "
+                f"Exception message: {str(e)}")
```

Comment:
I think we should probably just remove this and assume the actor will eventually start up or error. The error case will be handled in the `get_status` check.

Reply:
It could be helpful to surface error to user if the submission failed due to runtime_env errors, which won't populate to driver's stderr or visible by /logs API.